tests/test_metadata_serialization.py

```diff
@@ -50,18 +52,87 @@ def wrapper(test_cls: "TestSerialization"):
 
 class TestSerialization(unittest.TestCase):
 
+    invalid_signed: DataSet = {
+        "no _type": '{"spec_version": "1.0.0", "expires": "2030-01-01T00:00:00Z", "meta": {"f.txt": {"version": 1}}}',
+        "no spec_version": '{"_type": "signed", "version": 1, "expires": "2030-01-01T00:00:00Z", "meta": {"f.txt": {"version": 1}}}',
+        "no version": '{"_type": "signed", "spec_version": "1.0.0", "expires": "2030-01-01T00:00:00Z", "meta": {"f.txt": {"version": 1}}}',
+        "no expires": '{"_type": "signed", "spec_version": "1.0.0", "version": 1, "meta": {"f.txt": {"version": 1}}}',
+        "empty str _type": '{"_type": "", "spec_version": "1.0.0", "version": 1, "expires": "2030-01-01T00:00:00Z", \
+            "meta": {"f.txt": {"version": 1}}}',
+        "empty str spec_version": '{"_type": "signed", "spec_version": "", "version": 1, "expires": "2030-01-01T00:00:00Z", \
+            "meta": {"f.txt": {"version": 1}}}',
```

Comment:
Snapshot meta isn't required to contain anything: `"meta": {}` is enough if you're going for the minimal serialised form

nitpick on the style: I think this would be easier to read if you just barely need two lines (also allows names to be as long as you want):
```
        "empty str spec_version":
            '{"_type": "signed", "spec_version": "", "version": 1, "expires": "2030-01-01T00:00:00Z", "meta": {}}',
```

Reply:
Okay, will do that, but will add a comment that per spec having a snapshot object like this in a repository is considered invalid.